godot-core/src/obj/call_deferred.rs

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+use crate::builtin::{Callable, Variant};
+use crate::meta::UniformObjectDeref;
+use crate::obj::bounds::Declarer;
+use crate::obj::GodotClass;
+#[cfg(since_api = "4.2")]
+use crate::registry::signal::ToSignalObj;
+use godot_ffi::is_main_thread;
+use std::ops::DerefMut;
+
+// Dummy traits to still allow bounds and imports.
+#[cfg(before_api = "4.2")]
+pub trait WithDeferredCall<T: GodotClass> {}
+
+/// Trait that is automatically implemented for engine classes and  user classes containing a `Base<T>` field.
+///
+/// This trait enables type safe deferred method calls.
+///
+/// # Usage
+///
+/// ```no_compile
```

Comment:
I'd like to use a realisitic example here, but Engine classes are not available in core docs

Reply:
Are you sure about this? Prelude is just re-exporting other symbols, as long as they're in `godot-core` I don't see how that would matter...

What _might_ matter is the "minimal set" I mentioned above. If you can find an example using classes [in that list](https://github.com/godot-rust/gdext/blob/18d8a2200cb0bb85b587336ba853db3bb54c9ffa/godot-codegen/src/special_cases/codegen_special_cases.rs#L134-L134), that might fix it.

Reply:
Forgot to commit this one. Switched to Node2D. Works and makes sense as an example, i think